### ISSUE TITLE
Unable to load application: ArgumentError: invalid byte sequence in US-ASCII

### ISSUE THREAD

**Issue by andrewhavens** [puma/puma-dev#115]
I'm not sure what I changed, but all of a sudden, my app fails to boot using puma-dev. I have restarted puma-dev, but I'm still getting this error. I don't get this error when I start my app manually with `bundle exec puma`. Since there's no app specific backtrace, I really don't know what else to do to solve this.

```
! Booting app 'my-app' on socket /Users/andrew/.puma-dev/my-app/tmp/puma-dev-434.sock
my-app[44812]: bash: no job control in this shell
my-app[44812]: Puma starting in single mode...
my-app[44812]: * Version 2.14.0 (ruby 2.3.3-p222), codename: Fuchsia Friday
my-app[44812]: * Min threads: 0, max threads: 5
my-app[44812]: * Environment: development
my-app[44812]: ! Unable to load application: ArgumentError: invalid byte sequence in US-ASCII
my-app[44812]: bundler: failed to load command: puma (/Users/andrew/.rbenv/versions/2.3.3/bin/puma)
my-app[44812]: ArgumentError: invalid byte sequence in US-ASCII
my-app[44812]:   /Users/andrew/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rack-2.0.1/lib/rack/builder.rb:36:in `[]'
my-app[44812]:   /Users/andrew/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rack-2.0.1/lib/rack/builder.rb:36:in `parse_file'
my-app[44812]:   /Users/andrew/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/puma-2.14.0/lib/puma/configuration.rb:143:in `load_rackup'
my-app[44812]:   /Users/andrew/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/puma-2.14.0/lib/puma/configuration.rb:96:in `app'
my-app[44812]:   /Users/andrew/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/puma-2.14.0/lib/puma/runner.rb:113:in `load_and_bind'
my-app[44812]:   /Users/andrew/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/puma-2.14.0/lib/puma/single.rb:79:in `run'
my-app[44812]:   /Users/andrew/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/puma-2.14.0/lib/puma/cli.rb:215:in `run'
my-app[44812]:   /Users/andrew/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/puma-2.14.0/bin/puma:10:in `<top (required)>'
my-app[44812]:   /Users/andrew/.rbenv/versions/2.3.3/bin/puma:22:in `load'
my-app[44812]:   /Users/andrew/.rbenv/versions/2.3.3/bin/puma:22:in `<top (required)>'
! Killing 'my-app' (44812)
* App 'my-app' shutdown and cleaned up
! Detecting app 'my-app' dying on start
```

**Comment by MikeRogers0** [comment]
I ran into this problem also, I think it ended up being caused by german umlauts (such as ä ö ü ).

**Comment by evanphx** [comment]
Is this still an issue?

**Comment by andrewhavens** [comment]
@evanphx I eventually found that I had some special character (white space, maybe?) in my code. Not sure how it ended up in there but once I tracked it down and removed it, my issue was resolved. Not sure if puma-dev can help with this process by providing the filename and line number that is causing the problem. In general, sometimes I have a hard time tracking down issues when working with puma-dev because the error/backtrace does not fully bubble up.
